src/main.rs

```diff
@@ -57,11 +63,58 @@ pub struct Args {
     log_file: Option<String>,
 }
 
+/// Subcommands for 'qmassa'
+#[derive(Subcommand, Debug)]
+pub enum Commands{
+    /// Plots charts from JSON data
+    Plot {
+        /// Input JSON file
+        #[arg(short, long)]
+        input: String,
+
+        /// Output PNG file
+        #[arg(short, long)]
+        output: String,
+
+        /// Optional comma-separated list of charts to be plotted
+        #[arg(short, long, required = false)]
+        charts: Option<String>,
+    },
+}
+
 fn main() -> Result<()>
 {
     // parse command-line args
     let args = Args::parse();
 
+    // If a subcommand is provided, handle it
+    if let Some(subcommand) = &args.subcommand {
+        match subcommand {
+            Commands::Plot { input, output, charts } => {
```

Comment:
I think the content from this match can become another run_plot_cmd() function (like the other 2 now there).